### PR TITLE
Supplemental figure 1A: benchmarking time and memory usage

### PR DESCRIPTION
**Pull request by allyhawkins**
Related to #4 

This PR adds the first part of the benchmarking figure and includes a panel that looks at total run time and peak memory usage between Cell Ranger and Alevin-fry. To create these figures, I re-ran 6 samples (3 cell, 3 nuclei) through both methods using the same index, Ensembl v104. I added the `-with-trace` command to get the trace file with run time and memory information. 

- I added the trace files that I'm getting this information from to the repo. They're really small so it didn't make sense to put on S3, and I think we should keep them somewhere central so we don't lose them. There's one for the run of Cell Ranger and one for the run of Alevin-fry. 
- For wrangling the trace files, I took a lot of the code from our original benchmarking notebook (https://github.com/AlexsLemonade/alsf-scpca/blob/main/analysis/quantifier-comparisons/03-memory_time_comparison.Rmd). I simplified it and cleaned it up, because wow that code is _wild_. 
- I tried a few different approaches for the plot itself and I ended on a box plot with a sina plot... I know there is a lot of dislike towards boxplots, but with this few samples, I personally found it more visually appealing. I tried just a sina plot with a median line and a sina with violin and the violins just looked odd to me. So I'm curious what others think about the look here. 
- I chose to keep all single-cell and all single-nuclei together, instead of separating them. It just looked so sparse with only 3 samples per condition and they all are similar anyways. 
- These are my favorite colors I've ever used in a plot, but I know it might be hard to see the dots 😢 I can change to have two lighter colors, if people are having trouble. I am planning on using whatever colors we pick here in the rest of the benchmarking figure. 
- I also am wondering if it's okay to use the same type of plot here. I know the axes are different and they are technically measuring different things, but these different things are pretty related. 

Same deal as before, requesting @jaclyn-taroni for aesthetics and @jashapiro for aesthetics and code.

### CODE REVIEW

File: scripts/fig_s1a-memory-time-comparison.R [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+palette <- readr::read_tsv(palette_file)
+
+# get list of all colors 
+method_colors <- palette$color |> 
+  purrr::set_names(palette$method)
+
+
+# compare run time in minutes
+time_plot <- ggplot(grouped_log_df, aes(x = method, y = total_time, fill = method)) +
+  geom_boxplot(outlier.shape = NA) +
```
Comment (jashapiro) [inline scripts/fig_s1a-memory-time-comparison.R:96]: For the plot I was thinking something like: 
```
ggplot(grouped_log_df, aes(x = run_id, y = total_time, fill = method) + 
 geom_bar(position="dodge")
```

File: figures/pngs/FigS1A-time-memory-benchmarking.png
Comment (jaclyn-taroni) [inline figures/pngs/FigS1A-time-memory-benchmarking.png:1]: > With only 5 points, I wonder if we want a whole box plot, or maybe just showing the mean and the points. Part of me also wants to connect the points between samples run under each method, but I expect that will be busy.

I agree with no box plot for this number of samples and will add: what if each sample got its own shape as an alternative to connecting points?

Comment (allyhawkins) [reply figures/pngs/FigS1A-time-memory-benchmarking.png:1]: > Edited to add: To potentially head off a question here, I don't think the colors are really adding that much information. Edited, again: I suppose it depends on the formatting for the rest of the benchmarking figures.

If we keep the same format of the rest of the figures that we have in the [FAQ](https://scpca.readthedocs.io/en/stable/faq.html#why-did-we-use-alevin-fry-for-processing), then the colors will be used. 

> I agree with no box plot for this number of samples and will add: what if each sample got its own shape as an alternative to connecting points?

I like the idea of trying different shapes!

**Comment by jashapiro** [comment]
I like the shapes and bars, I think... I'd remove the SCPCR legends from all of them though. And for the bars I would order by time?

**Comment by jaclyn-taroni** [comment]
If we do the bars, that opens up the possibility to facet by single-cell vs. single-nuclei.

**Comment by allyhawkins** [comment]
What about something like this? I ordered by time and memory and then faceted by seq unit. @jaclyn-taroni @jashapiro 

image

**Comment by allyhawkins** [comment]
Okay, I've adjusted the code to now have the bar plot pictured above with some slight aesthetic fixes. @jaclyn-taroni let me know if you have any additional comments and @jashapiro this should be ready for review now.

**Review by jashapiro — APPROVED** [review]
LGTM

File: palettes/method-palette.tsv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+method	color
+Alevin-fry	#00274C
+Cell Ranger	#FFCB05
```
[the remark this answers is not in the document]
Comment (allyhawkins) [reply palettes/method-palette.tsv:3]: No way.

**Pull request merged**